subprojects/packagefiles/libarchive/tar/meson.build

```diff
@@ -0,0 +1,22 @@
+src = [
```

Comment:
use files().

Reply:
FWIW, the `meson rewrite` does not use files in those cases.

Reply:
files() helps with muon's formatter.

Reply:
muon's formatter is capable of discerning that order is not significant for arguments to the files() function, but order *might* be significant in arbitrary arrays.

This allows muon to alphanumerically sort arguments. Assuming you care. I don't care one way or another, myself.

Reply:
I should say that it's *arguably* not significant. There's no technical reason why people cannot use a list of `srcs = files(...)`, and then use `srcs[0]` for one specific executable and other indexes for other targets.

It is just that it's a reasonably safe assumption that doing so is quite unusual, so while technically sorting them is semantically changing the build file and thus unsafe, it's likely safe to do so in the common case. Whereas arrays are inherently not particularly likely to have files in them in the first place.